File: LeetCode/src/1614.py
class Solution:
    def maxDepth(self, s: str) -> int:
        depth, max_depth = 0, 0
        for ch in s:
            if ch == '(':
                depth += 1
            elif ch == ')':
                max_depth = max(depth, max_depth)
                depth -= 1
        return max_depth

File: LeetCode/src/1614.js
var maxDepth = function(s) {
    let depth = 0;
    let max_depth = 0;
    for (let ch = 0; ch < s.length; ch++) {
        if (s[ch] == '(') {
            depth += 1;
        } else if (s[ch] == ')'){
            max_depth = Math.max(max_depth, depth);
            depth -= 1;
        }
    }
    return max_depth;
};
